In [3]:
import os
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns

path = os.getcwd()

folder = ['11','12','13','14','15','16','17','18','19']
for pick in folder: 
    files = os.listdir(pick)
    df = pd.DataFrame(columns=('station', 'year', 'month', 'day', 'hour', 'air_temp', 'dew_pt_temp', 'sea_lvl_pressure', 
                           'wind_dirc', 'wind_spd', 'sky_con', 'l_precip_1', 'l_precip_6'))
    for f in files:
        print f
        i = 0
        station = f[:6]
        f = open(pick+'/'+f, "r")
        text = f.readlines()
        for line in text:
            line = line.strip('\n')
            line = line.split(' ')
            line = filter(None, line)
            df.loc[i, 'station'] = station
            df.loc[i, 1:] = [n for n in line]
            i += 1
    df.to_csv(pick+'.csv', sep=',')

722897-93206-2012
722897-93206-2013
722897-93206-2014
722897-93206-2015
722897-93206-2016
722950-23174-2012
722950-23174-2013
722950-23174-2014
722950-23174-2015
722950-23174-2016
723840-23155-2012
723840-23155-2013
723840-23155-2014
723840-23155-2015
723840-23155-2016
723890-93193-2012
723890-93193-2013
723890-93193-2014
723890-93193-2015
723890-93193-2016
724830-23232-2012
724830-23232-2013
724830-23232-2014
724830-23232-2015
724830-23232-2016
724926-23258-2012
724926-23258-2013
724926-23258-2014
724926-23258-2015
724926-23258-2016
724945-23293-2012
724945-23293-2013
724945-23293-2014
724945-23293-2015
724945-23293-2016
994027-99999-2012
994027-99999-2013
994027-99999-2014
994027-99999-2015
994027-99999-2016
998479-99999-2012
998479-99999-2013
998479-99999-2014
998479-99999-2015
998479-99999-2016


In [49]:
df.shape

(8783, 13)

import os
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns

path = os.getcwd()
files = os.listdir('wea_data')

df = pd.DataFrame(columns=('station', 'year', 'month', 'day', 'hour', 'air_temp', 'dew_pt_temp', 'sea_lvl_pressure', 
                           'wind_dirc', 'wind_spd', 'sky_con', 'l_precip_1', 'l_precip_6'))
f = open('722897-93206-2016', "r")

i = 0
#f = open(f, "rt")
station = '722897'
text = f.readlines()
for line in text:
        line = line.strip('\n')
        line = line.split(' ')
        line = filter(None, line)
        df.loc[i, 'station'] = station
        df.loc[i, 1:] = [n for n in line]
        i += 1
df.head()

In [57]:
df.shape

(8783, 13)

In [62]:
df.to_csv('2016722897222.csv', sep=',')